kornia/color/lab.py

```diff
@@ -37,27 +38,20 @@ def rgb_to_lab(image: torch.Tensor) -> torch.Tensor:
         raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
                          .format(image.shape))
 
-    # Convert from Linear RGB to sRGB
-    r: torch.Tensor = image[..., 0, :, :]
-    g: torch.Tensor = image[..., 1, :, :]
-    b: torch.Tensor = image[..., 2, :, :]
+    # Convert from sRGB to Linear RGB
+    lin_rgb = rgb_to_linear_rgb(image)
 
-    rs: torch.Tensor = torch.where(r > 0.04045, torch.pow(((r + 0.055) / 1.055), 2.4), r / 12.92)
-    gs: torch.Tensor = torch.where(g > 0.04045, torch.pow(((g + 0.055) / 1.055), 2.4), g / 12.92)
-    bs: torch.Tensor = torch.where(b > 0.04045, torch.pow(((b + 0.055) / 1.055), 2.4), b / 12.92)
-
-    image_s = torch.stack([rs, gs, bs], dim=-3)
-
-    xyz_im: torch.Tensor = rgb_to_xyz(image_s)
+    xyz_im: torch.Tensor = rgb_to_xyz(lin_rgb)
 
     # normalize for D65 white point
     xyz_ref_white = torch.tensor(
         [0.95047, 1., 1.08883], device=xyz_im.device, dtype=xyz_im.dtype)[..., :, None, None]
     xyz_normalized = torch.div(xyz_im, xyz_ref_white)
 
-    power = torch.pow(xyz_normalized, 1 / 3)
+    threshold = 0.008856
+    power = torch.pow(xyz_normalized.clamp_min(threshold), 1 / 3.)
```

Comment:
not whether `clamp_min` will be supported in older versions. I guess we could alternatively use `.clamp(min=threshold)`

Reply:
I'm a slave to copy/paste 🤣 replaced all with clip